File: backend/SindhDormitory.Domain/Entities/AuditLog.cs
// =============================================================================
// Domain/Entities/AuditLog.cs
// =============================================================================
using SindhDormitory.Domain.Enums;

namespace SindhDormitory.Domain.Entities;

public class AuditLog
{
    public long        LogId       { get; set; }
    public string      TableName   { get; set; } = string.Empty;
    public string      RecordId    { get; set; } = string.Empty;  // String to handle any PK type
    public AuditAction Action      { get; set; }
    public string?     OldValues   { get; set; }  // JSON
    public string?     NewValues   { get; set; }  // JSON
    public int?        PerformedByUserId { get; set; }
    public DateTime    PerformedAt { get; set; } = DateTime.UtcNow;
    public string?     IpAddress   { get; set; }
    public string?     UserAgent   { get; set; }

    // Navigation
    public User? PerformedBy { get; set; }
}

File: backend-node/src/entities/AuditLog.ts
import { Entity, PrimaryGeneratedColumn, Column, ManyToOne, JoinColumn, CreateDateColumn } from 'typeorm';
import { User } from './User';

@Entity('AuditLogs')
export class AuditLog {
  @PrimaryGeneratedColumn({ name: 'LogId' })
  logId!: number;

  @Column({ name: 'TableName', type: 'varchar', length: 100 })
  tableName!: string;

  @Column({ name: 'RecordId', type: 'varchar', length: 50 })
  recordId!: string;

  @Column({ name: 'Action', type: 'varchar', length: 10 })
  action!: string;

  @Column({ name: 'OldValues', type: 'json', nullable: true })
  oldValues!: any;

  @Column({ name: 'NewValues', type: 'json', nullable: true })
  newValues!: any;

  @Column({ name: 'IpAddress', type: 'varchar', length: 45, nullable: true })
  ipAddress!: string | null;

  @Column({ name: 'PerformedByUserId', type: 'int', nullable: true })
  performedByUserId!: number | null;

  @CreateDateColumn({ name: 'PerformedAt' })
  performedAt!: Date;

  @ManyToOne(() => User, (u) => u.auditLogs, { nullable: true })
  @JoinColumn({ name: 'PerformedByUserId' })
  performedBy!: User | null;
}
